lib/redis.js

```diff
@@ -99,6 +99,18 @@ module.exports = class Redis {
     return this.cmd('del', arguments)
   }
 
+  async lock(key, field, ttl) {
```

Comment:
Neat use of redis to build a lock. Curious if DDB could function in the same way? e.g. using the `lsd` in ddb as the lock?

Reply:
Yeah, that's sort of what I'm thinking.  But we may just delete the cached-impression-urls in DDB to indicate we already processed them.

So this lock is a _bit_ redundant with that?  But my thought was that handling race conditions would be easier here with redis than DDB.  Then the ingest lambda doesn't have to deal with locking anything while it hits pingback urls or something.